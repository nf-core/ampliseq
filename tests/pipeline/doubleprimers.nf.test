nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    options "-profile test_doubleprimers,docker"
    tag "pipeline"

    test("DoublePrimers") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/overall_summary.tsv")).match("overall_summary_tsv") },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                  path("$outputDir/barrnap/rrna.bac.gff"),
                                  path("$outputDir/barrnap/rrna.euk.gff"),
                                  path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/cutadapt/cutadapt_summary.tsv")).match("cutadapt") },
                { assert snapshot(path("$outputDir/dada2/ASV_seqs.fasta"),
                                  path("$outputDir/dada2/ASV_table.tsv"),
                                  path("$outputDir/dada2/DADA2_stats.tsv"),
                                  path("$outputDir/dada2/DADA2_table.rds"),
                                  path("$outputDir/dada2/DADA2_table.tsv")).match("dada2") },
                { assert new File("$outputDir/qiime2/input/rep-seqs.qza").exists() }, 
                { assert new File("$outputDir/qiime2/input/table.qza").exists() },                  
                { assert snapshot(path("$outputDir/input/Samplesheet_double_primer.tsv")).match("input") },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                                  path("$outputDir/multiqc/multiqc_data/multiqc_cutadapt.txt")).match("multiqc") }                      
            )
        }
    }
}
