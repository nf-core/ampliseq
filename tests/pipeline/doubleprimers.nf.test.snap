{
    "input": {
        "content": [
            "Samplesheet_double_primer.tsv:md5,cd63754fefa6e844a758efb893660f97"
        ],
        "timestamp": "2023-04-27T18:15:22+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,10c6c5ee2a8f0d21c1427e527939bd67"
        ],
        "timestamp": "2023-04-27T18:15:22+0000"
    },
    "cutadapt": {
        "content": [
            "cutadapt_summary.tsv:md5,ba0452e3edde6300fd111ad0e5b8e206"
        ],
        "timestamp": "2023-04-27T18:15:22+0000"
    },
    "overall_summary_tsv": {
        "content": [
            "overall_summary.tsv:md5,f0dd8c3dc3f45b84a1e32f88aabe7d08"
        ],
        "timestamp": "2023-04-27T18:15:22+0000"
    },
    "dada2": {
        "content": [
            "ASV_seqs.fasta:md5,bde6052256e9994a8ba6d4dd45d992e3",
            "ASV_table.tsv:md5,223eb043a32a1ce932e2e515c00125e6",
            "DADA2_stats.tsv:md5,89a8bbd8f229ba75268098078fb254de",
            "DADA2_table.rds:md5,e527269819e4056eed063dca9f25e3e7",
            "DADA2_table.tsv:md5,d30d5d392665181abddcf8ba5bdda102"
        ],
        "timestamp": "2023-04-27T18:15:22+0000"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,fbe6bb4c5527c0099577af622e35af67",
            "rrna.bac.gff:md5,05ddde74c06bb7377f2087a90e3ee1c3",
            "rrna.euk.gff:md5,df19e1b84ba6f691d20c72b397c88abf",
            "rrna.mito.gff:md5,04be33c4213eb82f66cf0652fe9e32d8"
        ],
        "timestamp": "2023-04-27T18:15:22+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_general_stats.txt:md5,8429be0a16adf09b6634bf31b430bfac",
            "multiqc_cutadapt.txt:md5,e89359b4478ef5d10620709f651f26a2"
        ],
        "timestamp": "2023-04-27T18:15:22+0000"
    }
}